src/Symfony/Component/FeatureFlags/FeatureChecker.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlags;
+
+final class FeatureChecker implements FeatureCheckerInterface
+{
+    public function __construct(
+        private readonly FeatureCollection $features,
+        private readonly bool $whenNotFound,
+    ) {
+    }
+
+    public function isEnabled(string $featureName): bool
```

Comment:
passing by, wanted to ask if it may be relevant to have an optional $context = [] passed here?
so implementation of each feature flag can leverage this and do more computation to decide true|false